src/Tasks/Copy.cs

```diff
@@ -269,7 +267,14 @@ private void LogDiagnostic(string message, params object[] messageArgs)
             if (OverwriteReadOnlyFiles)
             {
                 MakeFileWriteable(destinationFileState, true);
-                destinationFileExists = destinationFileState.FileExists;
+            }
+
+            // If the destination file is a hard or symbolic link, File.Copy would overwrite the source.
+            // To prevent this, we need to delete the existing entry before we Copy or create a link.
+            // We could try to figure out if the file is a link, but I can't think of a reason to not simply delete it always.
+            if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_6) && destinationFileState.FileExists && !destinationFileState.IsReadOnly)
```

Comment:
Should we also take into account OverwriteReadOnlyFiles and also delete them in that case?

Reply:
if `OverwriteReadOnlyFiles ` is set, the file is made writable and hence deleted here.

Reply:
...and now I see it 6 lines up. Stupid question, thanks!